### PR TITLE
[dhctl] Wait for resources required by manifest being created

### PR DESCRIPTION
**Pull request by mvasl**
## Description

Added a retry for resource creation for cases where resources required for manifest creation is not there yet

## Why do we need it, and what problem does it solve?

We hit that case where a resource specified in `config.yml` during bootstrap was meant to be created in a module namespace. As modules are initialized only after bootstrap process is complete, it deadlocked the installer. This allows the installer to wait for such missing resources.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: dhctl
type: fix
summary: Wait for resources required by manifest being created.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by mvasl** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.27 10109382394 10109382394-1-con-1-27 /sys/deckhouse-oss/install:pr9157

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by mvasl** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.27 10110090546 10110090546-1-con-1-27 /sys/deckhouse-oss/install:pr9157 redos@89.169.142.2

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've started the workflow for label 'e2e/run/yandex-cloud'...

File: dhctl/pkg/kubernetes/actions/resources/resources.go [hunk 1]
```diff
@@ -128,6 +132,37 @@ func (c *Creator) createAll() error {
 	return nil
 }
 
+func (c *Creator) ensureRequiredNamespacesExist() error {
+	knownNamespaces := make(map[string]struct{})
+
+	for _, res := range c.resources {
+		nsName := res.Object.GetNamespace()
+		if _, nsWasSeenBefore := knownNamespaces[nsName]; nsWasSeenBefore || nsName == "" {
```
Comment (name212) [inline dhctl/pkg/kubernetes/actions/resources/resources.go:140]:
```suggestion
		if _, nsWasSeenBefore := knownNamespaces[nsName]; nsName == ""  || nsWasSeenBefore {
```

**Review by aleksey-su — APPROVED** [review]

**Review by name212 — APPROVED** [review]

**Pull request merged**